### PR TITLE
aravis 0.6.0 (new formula)

### PR DESCRIPTION
**Pull request by hiszpanski**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Aravis is a library for interfacing with industrial cameras adhering to the GenICam / GigE Vision standards.

### CODE REVIEW

File: Formula/aravis.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  def install
+    inreplace "viewer/Makefile.am", "gtk-update-icon-cache", "gtk3-update-icon-cache"
+    system "./autogen.sh", "--disable-dependency-tracking",
+                           "--disable-silent-rules",
+                           "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/arv-tool-0.6"
```
Comment (fxcoudert) [inline Formula/aravis.rb]: We need a test that exercises the some of the functionality of the app. Is there absolutely nothing we can do without graphical interface?

Comment (thinkski) [reply Formula/aravis.rb]: One possibility is attempting to load the gstreamer plugin via `gst-inspect-1.0 aravissrc` -- this is a failure point if an older version of gstreamer (<= 1.12) is installed. Another option is to pass `--enable-cpp-test` to `./autogen.sh` to build a test program and run that, which may exercise the library more extensively than simply running `arv-tool-0.6`. Thoughts?

Comment (fxcoudert) [reply Formula/aravis.rb]: gstreamer plugin sounds good

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/aravis.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  def install
+    inreplace "viewer/Makefile.am", "gtk-update-icon-cache", "gtk3-update-icon-cache"
+    system "./autogen.sh", "--disable-dependency-tracking",
+                           "--disable-silent-rules",
+                           "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "gst-inspect-1.0 #{lib}/gstreamer-1.0/libgstaravis.0.6.so"
```
[the remark this answers is not in the document]
Comment (thinkski) [reply Formula/aravis.rb]: I can make this change, but it feels more fragile to check for the string. The output is:
```
Plugin Details:
  Name                     aravis
  Description              Aravis Video Source
  Filename                 /usr/local/lib/gstreamer-1.0/libgstaravis.0.6.so
  Version                  0.6.0
  License                  LGPL
  Source module            aravis
  Binary package           aravis
  Origin URL               http://blogs.gnome.org/emmanuel

  aravissrc: Aravis Video Source

  1 features:
  +-- 1 elements
```
Perhaps the filename location could differ with customized Homebrew installs? Test string would also need to be updated with each version update. As is, the test checks the return code of loading the plugin -- if the plugin does not load for any reason, i.e. the plugin is "blacklisted" by gstreamer, the return code is non-zero and the test fails. Please let me know if you'd still like me to make the change.

[the remark this answers is not in the document]
Comment (fxcoudert) [reply Formula/aravis.rb]: Can you `assert_match /Description *Aravis Video Source/, output`?

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @thinkski for the pull request!

**Pull request closed without merging**